lib/cfe_internal_hub.cf

```diff
@@ -110,6 +110,108 @@ bundle agent cfe_internal_hub_maintain
       action => if_elapsed_day;
 }
 
+bundle agent cfe_internal_hub_maintain_notice_retention_settings_change
+# @brief Emits reports to inform users that configuration settings have been
+# modified and need to be migrated to new location.
+#
+# Database maintenance process of purging of old data is now the responsibility
+# of cf-remote. This bundle looks for user modifications of default retention
+# and informs they should migrate their configuration.
+{
+  classes:
+    "running_binary_with_new_db_retention_settings"
+      expression => "cfengine_3_22_0|feature_cf_reactor_report_retention";
```

Comment:
This was just a WAG for the context. Ideally, we will have a non version based class to key off of for this.

Reply:
Yeah, maybe a feature class? Or something similar.